hw/ip/rstmgr/data/rstmgr.sv.tpl

```diff
@@ -217,10 +241,61 @@ module rstmgr import rstmgr_pkg::*; (
   % endfor
 % endfor
 
+  ////////////////////////////////////////////////////
+  // Crash info capture                             //
+  ////////////////////////////////////////////////////
+  localparam int CrashRemainder = $bits(alert_pkg::alert_crashdump_t) % RdWidth > 0 ? 1 : 0;
+  localparam int CrashStoreSlot = $bits(alert_pkg::alert_crashdump_t) / RdWidth +
+      CrashRemainder;
+  localparam int TotalWidth     = CrashStoreSlot * RdWidth;
+  localparam int SlotCntWidth   = $clog2(CrashStoreSlot);
+
+  logic dump_capture;
+  logic [2**SlotCntWidth-1:0][RdWidth-1:0] slots;
+  logic [CrashStoreSlot-1:0][RdWidth-1:0] slots_q;
+
+  // capture on any legal reset request
+  assign dump_capture = reg2hw.alert_info_ctrl.en.q &
+                        (rst_hw_req | rst_ndm | rst_low_power);
+
+  if (2**SlotCntWidth > CrashStoreSlot) begin: gen_tieoff_arrays
+    assign slots[2**SlotCntWidth-1:CrashStoreSlot] = '0;
+  end
```

Comment:
if you want, this could be abbreviated a bit like this:
```systemverilog
always_comb begin : p_output_assign
  slots = '0;
  slots = slots_q;
end
```

Reply:
ah yes, i should change to that.

Reply:
huh didn't know that I don't have to specify the bitwidth for `slots`.

Reply:
i feel for the second assignment `slots = slots_q` i still have to specify which ones no..? 
I'll give it a try.